net: wifi: Current max rate and over run statistics

kapbh opened the pull request:
Current PHY rate
It represents the current PHY rate of transfer of data in bits per second. It will a TX data rate.

Over run count
It represents the number of packets dropped either at received and sent due to lack of buffer memory to retain all packets on the network interface.

jukkar commented on include/zephyr/net/net_stats.h:
```diff
@@ -623,6 +623,12 @@ struct net_stats_wifi {
 	/** Total number of unicast packets received and sent */
 	struct net_stats_pkts unicast;
+
+	/** The current maximum PHY rate of transfer of data*/
+	net_stats_t current_max_rate;
```
This does not feel very statistics like variable. Would using #76062 give better real time information for the rate?

krish2718 replied:
`wifi_iface_status` is a better place for this, it's mostly depends on `enum wifi_link_mode link_mode;` and capabilities of the peer.

krish2718 commented on include/zephyr/net/net_stats.h:
```diff
@@ -623,6 +623,9 @@ struct net_stats_wifi {
 	struct net_stats_pkts unicast;
+
+	/** Total number of dropped packets at received and sent*/
+	net_stats_t overrun_count;
```
please split this into two commits.

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -522,6 +522,8 @@ struct wifi_iface_status {
 	/** is TWT capable? */
 	bool twt_capable;
+	/** The current maximum PHY rate of transfer of data */
+	int current_max_rate;
```
`current_max_phy_rate`

kapbh committed:
net: wifi: Add over run count statistics

Over run count
It represents the number of packets dropped either at
received and sent due to lack of buffer memory to retain
all packets on the network interface.

Signed-off-by: Kapil Bhatt <kapil.bhatt@nordicsemi.no>

kapbh committed:
net: wifi: Add Current PHY rate

Current PHY rate
It represents the current PHY rate of transfer
of data in bits per second. It will a TX data rate.

Signed-off-by: Kapil Bhatt <kapil.bhatt@nordicsemi.no>

krish2718 commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -877,6 +878,7 @@ static void print_wifi_stats(struct net_if *iface, struct net_stats_wifi *data,
 	PR("Beacons missed   : %u\n", data->sta_mgmt.beacons_miss);
 	PR("Unicast received : %u\n", data->unicast.rx);
 	PR("Unicast sent     : %u\n", data->unicast.tx);
+	PR("Over run count   : %u\n", data->overrun_count);
```
```suggestion
	PR("Overrun count   : %u\n", data->overrun_count);
```

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -522,6 +522,8 @@ struct wifi_iface_status {
 	unsigned short beacon_interval;
 	/** is TWT capable? */
 	bool twt_capable;
+	/** The current maximum PHY rate of transfer of data */
```
```suggestion
	/** The current maximum 802.11 PHY data rate */
```
